code/network/multi_pxo.cpp

```diff
@@ -421,28 +398,27 @@ const char *Multi_pxo_player_slider_name[GR_NUM_RESOLUTIONS] = {
 	"2_slider"			// GR_1024
 };
 
-// head of the list of players in this channel
-player_list *Multi_pxo_players = NULL;
-int Multi_pxo_player_count = 0;
+// the list of players in the current channel
+SCP_vector<SCP_string> Multi_pxo_players_vec;
```

Comment:
The whole point of this process is to be able to index these arrays for the API. Looking at various implementations of SCP_list in the code, it doesn't look like that will work with ADE_INDEXER either. Is there a way to use it in the same way?

Reply:
Hmm, good question. It's used enough that I just assumed it would work for this, but perhaps I'm wrong.

Thinking about it more I believe that using vector here might be okay. It's a horrible use case for this, and the chat lines too for that matter, but the limited vector size likely makes the performance hit a non-issue. Max clients in a chat channel is 50, so the list of players will never get larger than that.

So unless someone else wants to chime in here then perhaps you should just leave it as is. If it becomes a problem later then perhaps we can revisit it at that point.

Reply:
The other way I think think to get the data through the API is with a single ADE_FUNC that returns the entire list as a table. The FUNC would iterate through the entire list to build the table before sending it. That feels similarly inefficient but perhaps it comes down to how often the list should be updated since, as you said, we're not talking about a huge amount of data here. Do you know off the top of your head how often that happens?

I do want to get this right because there are other similar lists that will need work as well in later multi UIs.

Reply:
It's just going to update whenever a player comes and goes. Sometimes that will happen pretty quickly, sometimes it's going to be a while. And the same player can come and go multiple times in the span of a couple of minutes if they're looking for games. There's just no telling really.

I checked the stats on PXO and the FSO chat server hasn't had more than 4 concurrent users for at least the past few months. So honestly, with these numbers, I'm not sure you're going to get a simpler or more efficient setup than just using a vector. It's a poor choice in general practice but for this specific case I'm finding it hard to think of a better option.

Up to you though, if you think you might come up a more appropriate solution.